ui/src/app/applications/components/utils.tsx

```diff
@@ -123,6 +124,74 @@ export async function deleteApplication(appName: string, appNamespace: string, a
     return confirmed;
 }
 
+export async function scalePods(resource: ResourceTreeNode, metadata: models.ObjectMeta, apis: ContextApis): Promise<boolean> {
```

Comment:
imo, if at all possible, the UI should generically present "parameters" communicated by the action definition itself. Hard-coding a parameters UI for just the scale action significantly limits the power of this feature.

Reply:
I believe I understand your point.
But we cannot use the current logic with discovery actions.
Currently for all actions which returned from backend by getResourceActions function is built default ui which triggers this action and show confirmation popup.
In our case we need custom UI with pop up which you can see in the PR body and this solution does not work.

Reply:
> we cannot use the current logic with discovery actions

The actions currently communicate up info like "disabled" or "not disabled."

I think we could augment that information with "has parameters," "list of parameters," "validation info for parameters," etc. The UI could react to that configuration and display the appropriate popup and inputs.

Reply:
@crenshaw-dev We can, but for scale case it is special and custom popup.
<img width="1124" alt="Screenshot 2023-09-19 at 12 01 17" src="https://github.com/argoproj/argo-cd/assets/29062574/c4d8a277-e474-46c4-928f-8fe293b98101">
How would you suggest to process this case

Reply:
Let's consolidate the discussion over here: https://github.com/argoproj/argo-cd/pull/15633#discussion_r1334669882

Reply:
@crenshaw-dev 
Hi
Did you have to take a look at my changes?